src/plugins/intel_cpu/src/nodes/strided_slice.cpp

```diff
@@ -38,8 +39,113 @@ bool StridedSlice::isSupportedOperation(const std::shared_ptr<const ov::Node>& o
     return true;
 }
 
+inline int calculate_output_shape(const int& startIdx, const int& stopIdx, const int& step, const int& len) {
+    auto start = (startIdx > len - 1) ? len - 1 : ((startIdx % len + len) % len);
+    auto stop = (stopIdx > len - 1)   ? len     : ((stopIdx % len + len) % len);
+    return (stop - start + step - 1) / step;
+}
+
+class StridedSliceShapeInfer : public ShapeInferEmptyPads {
+public:
+    StridedSliceShapeInfer(const std::shared_ptr<ov::Node>& op,
+                           const IShapeInfer::port_mask_t& port_mask,
+                           const std::set<int64_t>& begin_mask,
+                           const std::set<int64_t>& end_mask,
+                           const std::set<int64_t>& new_axis_mask,
+                           const std::set<int64_t>& shrink_axis_mask)
+    : m_op(op), m_port_mask(port_mask), m_outputShape(op->get_output_partial_shape(0).rank().get_length(), 1),
+      m_begin_mask_set(begin_mask), m_end_mask_set(end_mask), m_new_axis_mask_set(new_axis_mask), m_shrink_axis_mask_set(shrink_axis_mask) {}
```

Comment:
```suggestion
    StridedSliceShapeInfer(const std::shared_ptr<ov::Node>& op,
                           IShapeInfer::port_mask_t port_mask,
                           std::unordered_set<int64_t> begin_mask,
                           std::unordered_set<int64_t> end_mask,
                           std::unordered_set<int64_t> new_axis_mask,
                           std::unordered_set<int64_t> shrink_axis_mask)
    : m_op(op), m_port_mask(port_mask), m_outputShape(op->get_output_partial_shape(0).rank().get_length(), 1),
      m_begin_mask_set(std::move(begin_mask)),
      m_end_mask_set(std::move(end_mask)),
      m_new_axis_mask_set(std::move(new_axis_mask)),
      m_shrink_axis_mask_set(std::move(shrink_axis_mask)) {}
```
It is better to use hash table than a tree.
unordered_set has move constructor, then it is better to pass by value to avoid extra copy.

Reply:
> It is better to use hash table than a tree. unordered_set has move constructor, then it is better to pass by value to avoid extra copy.

accept and refresh it.